src/java.desktop/share/classes/javax/swing/text/DefaultEditorKit.java

```diff
@@ -339,7 +339,12 @@ public void write(Writer out, Document doc, int pos, int len)
                 for (int counter = last; counter < maxCounter; counter++) {
                     if (array[counter] == '\n') {
                         if (counter > last) {
-                            out.write(array, last, counter - last);
+                            if(array[counter-1] == '\r') {
```

Comment:
The space between if and the opening parenthesis is still missing.

The spaces around `-` are also missing.